tools/gm-to-obj/gm-to-obj.py

```diff
@@ -0,0 +1,371 @@
+        def add_vt(self, u, v):
+            self.vt += "vt %f %f\n" % (u, v)
+
+        # List of the faces
+        def add_f(self, x, y, z):
+            self.f += "f %d %d %d\n" % (x+1+self.svertex,
+                                        y+1+self.svertex, z+1+self.svertex)
+
+    # Dictionaries used to store objects' start and end
+    list_objects = []
+
+    # Get instructions from objects
+    print("Found objects:", len(robjects))
+    for obj in robjects:
+        list_objects.append(ProperObj(getattr(obj, "name"), getattr(
+            obj, "svertex"), getattr(obj, "striangle"), getattr(
+            obj, "nvertices"), getattr(obj, "ntriangles"), getattr(obj, "vertex_buff")))
+        # Start transforming to obj
+    print("\nVertex buffers:")
+
+    vertex_in_buffer = []
+    # Find all the vertices
+    for v in range(0, rhead.nvrtbuffs):
+        size = getattr(rvb[v], "size")
+        print("\t-Buffer", v, "has Type:", rvb[v].type)
+        vertex_type = get_vertex_from_wasted_bytes(
+            ((rvb[v].type & 3) + (rvb[v].type >> 2)) * 8)
+        stride = sizeof(vertex_type)
+        nverts = int(size/stride)
+        vertex_in_buffer.append(get_array_of(vertex_type.__name__, nverts))
+
+    # The start_vertex is wrong, as workaround it is calculated from scratch
+    workaround_s_vertex = 0
+    print("\nGenerating objects:")
+    for o in list_objects:
+        print("\t-", o.parsed_name)
+        v_buffer = vertex_in_buffer[o.vertex_buff]
+        for index in range(o.svertex, o.svertex+o.nvertices):
+            vert = v_buffer[index]
+            o.add_v(*getattr(getattr(vert, "pos"), "v"))
+            o.add_vn(*getattr(getattr(vert, "norm"), "v"))
+            o.add_vt(getattr(vert, "tu0"), getattr(vert, "tv0"))
+
+        o.svertex = workaround_s_vertex
+        workaround_s_vertex += o.nvertices
+
+        for index in range(o.striangle, o.striangle+o.ntriangles):
+            tri = rtriangles[index]
+            o.add_f(*getattr(tri, "vindex"))
+
+    # start crating obj file
+    print("\nWriting output               ")
+    out = open(output_name, "w")
+
+    out.write("# Obj generator for the storm-engine \n")
+    out.write("# Author: https://github.com/MangioneAndrea \n")
+    # Add smoothing
+    out.write("s 1 \n")
+
+    # Write everything to file
+    for obj in list_objects:
+        out.write(obj.o)
+        out.write(obj.v)
+        out.write(obj.vt)
+        out.write(obj.vn)
+        out.write(obj.f)
+
+    out.close()
+
+    print("\nDone")
+
+
+if (__name__ == "__main__"):
+    # define args
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "path", help="Where is the .gm file located?", type=str)
+    parser.add_argument("--output", "-o", help="Output name", type=str)
+    args = parser.parse_args()
+    args_input = str(args.path)
+    args_output = args.output
+    if(args_output is None):
+        args_output = ntpath.basename(args_input).replace(".gm", ".obj")
```

Comment:
I think output files should be written in the same directory as input files.
Running
`$ ./gm-to-obj.py ../../RESOURCE/MODELS/WorldMap/spanish_manowar.gm`
outputs a file in the current directory.